### PR TITLE
Fix crash in brave_ads migration

### PR DESCRIPTION
**Pull request by jonathanKingston**
Fixes https://github.com/brave/brave-browser/issues/8132

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

Upgrade from 1.5.x to 1.6.8, and confirm `DB: Error migrating database from v3 to v4` does not appear in the console logs

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_ads/browser/bundle_state_database.cc [hunk 1]
```diff
@@ -635,11 +635,7 @@ bool BundleStateDatabase::MigrateV2toV3() {
 
 bool BundleStateDatabase::MigrateV3toV4() {
   std::string sql = "ALTER TABLE ad_info ADD advertiser_id LONGVARCHAR;";
-  if (!GetDB().Execute(sql.c_str())) {
-    return false;
-  }
-
-  return false;
+  return GetDB().Execute(sql.c_str());
```
Comment (tmancey) [inline components/brave_ads/browser/bundle_state_database.cc]: Great work, thank you. For this PR could you also resolve `BundleStateDatabase::MigrateV1toV2` as the final `return false;` should also be `return true;`

Comment (jonathanKingston) [reply components/brave_ads/browser/bundle_state_database.cc]: Ah I guess I stated at a later migration in the first place to not hit that one. Perhaps that's related to the other crash reports I see in the tracker.

**Comment by tmancey** [comment]
@jonathanKingston great work and thank you for taking the time to resolve this issue. As soon as CI passes I will merge to `master`, keep up the great work!

**Comment by jonathanKingston** [comment]
@tmancey thanks, I was about to ask what the merge process was :)

**Comment by tmancey** [comment]
@jonathanKingston CI passed and merged with `master`. Once again thank you

**Pull request merged**